Mono.Debugging.Soft/PortablePdbData.cs

```diff
@@ -153,7 +162,7 @@ SourceLinkMap[] GetSourceLinkMaps ()
 
 		internal SoftScope [] GetHoistedScopesPrivate (MethodMirror method)
 		{
-			using (var fs = new FileStream (pdbFileName, FileMode.Open, FileAccess.Read, FileShare.Read))
+			using (var fs = stream)
 			using (var metadataReader = MetadataReaderProvider.FromPortablePdbStream (fs)) {
```

Comment:
Might be a dumb question here, but what happens if say GetSourceLinkMaps is called and then GetHoistedScopesPrivate is called? The stream is disposed by GetSourceLinkMaps - will that result in an exception? Just wondering since the original code creates a new file stream each time.

Reply:
@mrward from what I quickly saw from the code that uses it in Mono.Debugger is that the code pattern is

```
    ctx.Session.GetPdbData (variable.Method)?.XXXXX
```

where `GetPdbData` returns a new instance of `PortablePdbData` each time, so it does not look like we hit this but it does seem like a valid concern. Essentially we can use this object just the once if we hit a method that uses stream which doesn't sound entirely right @jstedfast @nosami

Reply:
Yeah, this code is not very good. I've found a better way to do the same thing, so I'm going to rewrite this part.

Reply:
The better way to do this isn't supported on latest Android, so going to need to refactor.

Reply:
The fact that it returns a new instance of `PortablePdbData` each time needs changing to return a cached version, so the stream issue would become a bigger concern

Reply:
Fixed